### ISSUE TITLE
Increase download speed with multi-connection request

### ISSUE THREAD

**Issue by MrT3acher** [EpicWink/proxpi#30]
It's so faster if change `_download_file` method to support multi-connection download.

https://github.com/EpicWink/proxpi/blob/caa09def52520a15583c350171111aa97f10bd4b/src/proxpi/_cache.py#L654

## Problem definition
Download is slow for my network situation. (I believe others can experience it, too.)

## Proposed solution (optional)
make `_download_file` method multi-connection.

**Comment by EpicWink** [comment]
Right now, the downloading of a file will first try to download from the source index, waiting some time (default up to 0.9 seconds) before returning the downloaded file to the client. If the file is not downloaded within that time, the client is redirected to download directly from the source index.

pip will then download the file, and move on to getting the next dependency (if any), which will start the process again. This means that on slow connections, there will already be multiple files being downloaded concurrently as both pip and `proxpi` the same file, and any future large files which are also too slow to be downloaded.

In summary, I don't think a parallelised single-file download will achieve much speedup. In addition, this increased load on PyPI is one of the main things I'm trying to avoid with `proxpi`

**Comment by MrT3acher** [comment]
The main problem is that in my network I can't access to pypi.org and its indexes directly. so I use proxpi as a proxy-cache server. so I changed `PROXPI_DOWNLOAD_TIMEOUT` to a very large number to avoid redirecting user to main pypi.org index url. but at this point proxpi downloads the python packages slowly. I tested downloading packages with a Download Manager and it's very fast (concurrent connection speeds up the download).

this feature can be controlled using some environment variable like `PROXPI_DOWNLOAD_CONNECTIONS` with default value equal to 1 that means no concurrency.

If you are OK, I can develop this part and make a merge request.

**Comment by EpicWink** [comment]
Feel free to implement this. If the default is to not performed a ranged request, I'm fine with that
